src/app/pages/index.component.ts

```diff
@@ -1,58 +1,27 @@
 import { Component, OnInit } from '@angular/core';
 
-import { ICourse } from '../shared/icourse';
 import { Course } from '../shared/course';
 import { CourseService } from '../shared/course.service';
+import { SearchPipe } from '../shared/search.pipe';
 
 
 @Component({
-  selector: 'courses-page',
-  templateUrl: './index.component.html'
+  selector: 'index-page',
+  templateUrl: './index.component.html',
+  styleUrls: ['./index.component.scss'],
+  providers:[SearchPipe]
 })
 export class IndexComponent {
   courses: Course[] = [];
+  private _pipe: SearchPipe = new SearchPipe();
```

Comment:
As far as I know, it's more preferable to import pipes using dependency injection

Reply:
ok, I got it.